string v_traj
string r_traj